pytorch_toolkit/ote/ote/modules/arg_parsers/custom_classes.py

```diff
@@ -26,13 +26,32 @@
 
 
 @ARG_PARSERS.register_module()
-class CustomDetectorArgParser(DefaultArgParser):
-    def __init__(self):
-        super(CustomDetectorArgParser, self).__init__()
+class CustomClassesArgParser(DefaultArgParser):
 
     def get_train_parser(self, config_path):
         parser = argparse.ArgumentParser(parents=[train_args_parser(config_path)], add_help=False)
         parser.add_argument('--classes', required=False,
                             help='Comma-separated list of classes (e.g. "cat,dog,mouse").')
 
         return parser
+
+    def get_test_parser(self, config_path):
+        parser = argparse.ArgumentParser(parents=[test_args_parser(config_path)], add_help=False)
+        parser.add_argument('--classes', required=False,
```

Comment:
2 @Ilya-Krylov     
Ilya, I think, `required=True` should be here (and below).
It is need to be sure that external app calls the script with the required number of classes.

As we considered, one of "fragile use cases" is when somebody instantiates a template, does not make any operations with it, and then copies IR to the folder of the template just to make evaluation.
In this case if the file `*.extra_params` (or as I proposed `*.extra_params.yml` ) is absent, the script `eval.py` will give a wrong result without any clarification. (This is what I mean by "fragility")

Obviously, at the moment you use another decision when the file `*.extra_params` reads **_always_**  if the parameter `--classes` is not passed, so it is impossible to run `eval.py` on an IR without `export.py` before.
But it seems to me that it would be better to make the file `*.extra_params`  optional,  but make the parameter `--classes` required.
Note that (AFAIK) the parameter `classes` should be always present for model templates of custom detection (if it is not present it is a bug, isn't it?), so it is not an issue to pass `--classes` to the train/eval/export/compress scripts.

Reply:
not relevant anymore, I guess